### PR TITLE
Feat/client zookeeper

### PR DESCRIPTION
**Pull request by Emily-Case**

### CODE REVIEW

File: source/table_client.c [hunk 1]
```diff
@@ -13,24 +13,72 @@
 #include "sdmessage.pb-c.h"
 #include "table_client.h"
 #include "stats.h"
+#include <zookeeper/zookeeper.h>
+#include "watcher_callbacks.h"
```
Comment (enderesting) [inline source/table_client.c:17]: put it in `table_client_client.h` maybe? for clarity

File: source/table_client.c [hunk 2]
```diff
@@ -13,24 +13,79 @@
...
+    while (!terminated && connected_to_head && connected_to_tail) {
+
+        zoo_string* new_children_list =	(zoo_string *) malloc(sizeof(zoo_string));
+
+        if (zoo_get_children(zh, zoo_root, 1, new_children_list) != ZOK) {
+            perror("Error getting child nodes\n");
+            exit(-1);
+        }
+
+        if(children_list->data != new_children_list->data) {
```
Comment (enderesting) [inline source/table_client.c:70]: I imagined you are trying to compare two lists are the same, but you can't compare pointers this way, the pointers themselves will always be different

Comment (enderesting) [reply source/table_client.c:70]: maybe try a for loop with`strcmp` instead?

File: source/table_client.c [hunk 3]
```diff
@@ -13,24 +13,79 @@
...
+            char* new_head_path = new_children_list->data[0];
+            char* new_tail_path = new_children_list->data[children_list->count-1];
+
+            if(head_path != new_head_path) {
```
Comment (enderesting) [inline source/table_client.c:74]: use `strcmp` here also

File: source/table_client.c [hunk 4]
```diff
@@ -13,24 +13,79 @@
...
+                rtable_disconnect(rtable_head);
+                strcpy(head_path, new_head_path);
+                zoo_get(zh, head_path, 0, zoo_data_head, zoo_data_len, NULL);
+                struct rtable_t* rtable_head = rtable_connect(zoo_data_head);
+            }
+
+            if(tail_path != new_tail_path) {
+                rtable_disconnect(rtable_tail);
+                strcpy(tail_path, new_tail_path);
+                zoo_get(zh, head_path, 0, zoo_data_tail, zoo_data_len, NULL);
```
Comment (enderesting) [inline source/table_client.c:84]: typo

**Review by enderesting — COMMENTED** [review]
Added comments

File: source/table_client.c (hunk 1 above)
Comment (Emily-Case) [reply source/table_client.c:17]: oh yeah, I meant to do so.

File: source/table_client.c (hunk 2 above)
Comment (Emily-Case) [reply source/table_client.c:70]: on it

File: source/table_client.c (hunk 3 above)
Comment (Emily-Case) [reply source/table_client.c:74]: on it

File: source/table_client.c (hunk 4 above)
Comment (Emily-Case) [reply source/table_client.c:84]: where is the typo, I must be blind bc I can't see it

**Review by enderesting — APPROVED** [review]

**Pull request merged**